wbb/modules/greetings.py

```diff
@@ -107,112 +107,119 @@ async def get_initial_captcha_cache():
-                )
+        )
+    for i in range(2, 5):
+        temp_keyboard_2.append(
+            InlineKeyboardButton(
+                f"{wrong_answers[i]}",
+                callback_data=f"pressed_button {wrong_answers[i]} {member.id}",
             )
-        for i in range(5, 8):
-            temp_keyboard_3.append(
-                InlineKeyboardButton(
-                    f"{wrong_answers[i]}",
-                    callback_data=f"pressed_button {wrong_answers[i]} {member.id}",
-                )
+        )
+    for i in range(5, 8):
+        temp_keyboard_3.append(
+            InlineKeyboardButton(
+                f"{wrong_answers[i]}",
+                callback_data=f"pressed_button {wrong_answers[i]} {member.id}",
             )
-
-        shuffle(temp_keyboard_1)
-        keyboard = [temp_keyboard_1, temp_keyboard_2, temp_keyboard_3]
-        shuffle(keyboard)
-        verification_data = {
-            "chat_id": message.chat.id,
-            "user_id": member.id,
-            "answer": captcha_answer,
-            "keyboard": keyboard,
-            "attempts": 0,
-        }
-        keyboard = InlineKeyboardMarkup(keyboard)
-        # Append user info, correct answer and
-        answers_dicc.append(verification_data)
-        # keyboard for later use with callback query
-        button_message = await message.reply_photo(
-            photo=captcha_image,
-            caption=text,
-            reply_markup=keyboard,
-            quote=True,
         )
-        os.remove(captcha_image)
 
-        # Save captcha answers etc in mongodb in case bot gets crashed or restarted.
-        await update_captcha_cache(answers_dicc)
-
-        asyncio.create_task(
-            kick_restricted_after_delay(WELCOME_DELAY_KICK_SEC, button_message, member)
-        )
-        await asyncio.sleep(0.5)
+    shuffle(temp_keyboard_1)
+    keyboard = [temp_keyboard_1, temp_keyboard_2, temp_keyboard_3]
+    shuffle(keyboard)
+    verification_data = {
+        "chat_id": message.chat.id,
+        "user_id": member.id,
+        "answer": captcha_answer,
+        "keyboard": keyboard,
+        "attempts": 0,
+    }
+    keyboard = InlineKeyboardMarkup(keyboard)
+    # Append user info, correct answer, and
+    answers_dicc.append(verification_data)
+    # keyboard for later use with callback query
+    button_message = await app.send_photo(
+        chat_id=chat.id,
+        photo=captcha_image,
+        caption=text,
+        reply_markup=keyboard,
+    )
+    os.remove(captcha_image)
+
+    # Save captcha answers etc in mongodb in case the bot gets crashed or restarted.
+    await update_captcha_cache(answers_dicc)
+
+    asyncio.create_task(
+        kick_restricted_after_delay(WELCOME_DELAY_KICK_SEC, button_message, member)
+    )
+    await asyncio.sleep(0.5)
 
+@app.on_message(filters.new_chat_members, group=welcome_captcha_group)
+@capture_err
+async def welcome(_, message: Message):
```

Comment:
this handler (with filters.new_chat_members) won't be called by pyrogram when the chat join request gets auto approved?

Did you properly check if handle_new_member() is not being called twice? (by welcome() and the manual call  in autoapprove.py at the end)

Reply:
Nope it won't. If you want you can verify https://t.me/+PR9SmmwO5GkwZDg1